ObjectivePGP/CryptoBox/PGPEC.m

```diff
@@ -268,7 +270,7 @@ + (BOOL)publicEncrypt:(nonnull NSData *)data withPublicKeyPacket:(PGPPublicKeyPa
             pgp_defer {
                 OPENSSL_clear_free(sigret, siglen);
             };
-            if (EVP_DigestSign(ctx, sigret, &siglen, toSign.bytes, toSign.length) <= 0) {
+            if (EVP_DigestSign(ctx, sigret, &siglen, hash.bytes, hash.length) <= 0) {
```

Comment:
I assume this is due to https://datatracker.ietf.org/doc/html/draft-ietf-openpgp-rfc4880bis-10#section-14.8

Reply:
Sure, OpenSSL documentation is a bit confusing, since the method is EVP_DigestXXX but in case of Ed25519 hash is not supported, parameter should be NULL. I have tested between iOS and Android (BouncyCastle) clients and the signature was problematic however the data from which the signature is generated was the same. After prehashing the data to be signed or verified the problem solved.